db/sequence_allocator.go

```diff
@@ -251,23 +253,26 @@ func (s *sequenceAllocator) nextSequenceGreaterThan(ctx context.Context, existin
 	s.dbStats.SequenceReservedCount.Add(int64(numberToRelease + numberToAllocate))
 	s.dbStats.SequenceAssignedCount.Add(1)
 
-	// Release previously allocated sequences (c), if any
-	err = s.releaseSequenceRange(ctx, prevAllocReleaseFrom, prevAllocReleaseTo)
-	if err != nil {
-		base.WarnfCtx(ctx, "Error returned when releasing sequence range [%d-%d] for previously allocated sequences. Will be handled by skipped sequence handling.  Error:%v", prevAllocReleaseFrom, prevAllocReleaseTo, err)
+	if prevAllocReleaseFrom < prevAllocReleaseTo {
```

Comment:
Shouldn't this be prevAllocReleaseFrom <= prevAllocReleaseTo?  Beyond that, I think this condition is already handled inside releaseSequenceRange.  Given that this is a logging enhancement it would be good to avoid changes that could introduce new regressions, unless you spotted a scenario that was previously being mishandled.

Reply:
Good catch, this was from earlier when I realized that `prevAllocReleaseTo` could be -1.